Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm

```diff
@@ -853,9 +883,59 @@ - (IBAction)_popUpButtonAction:(id)sender
         break;
     }
 
-    inspectedViewFrameDidChange(currentDimension);
+    NSView *inspectedViewParent = [inspectedView superview];
+    NSView *inspectorViewParent = inspectedViewParent;
+
+    if (inspectedView.translatesAutoresizingMaskIntoConstraints) {
+        // Impose a manual resize on the inspected view based on the inspector's size
+        inspectedViewFrameDidChange(currentDimension);
+    } else {
+        // Because the inspected view may be laid out using constraints or autosizing mask, but
+        // only autosizing mask is compatible with sharing space with the inspector view, take a
+        // different approach when the inspected view appears to be laid out by constraints.
+
+        // Put the inspected view and the inspector view in a shared container, and set the shared container
+        // up to possess the layout attributes of the original inspected view.
+        RetainPtr containerView = adoptNS([[NSView alloc] initWithFrame:[inspectedView frame]]);
+        containerView.get().autoresizingMask = inspectedView.autoresizingMask;
+        containerView.get().translatesAutoresizingMaskIntoConstraints = NO;
+        inspectorViewParent = containerView.get();
+
+        // Impose a manual resize on the inspected view based on the inspector's size
+        inspectedViewFrameDidChange(currentDimension);
+
+        // Transfer any constraints involving the inspectedView so that they relate instead to the containerView
+        NSArray *transferredConstraints = transferConstraintsAwayFromInspectedView(inspectedView.window.contentView, inspectedView, containerView.get()).get();
```

Comment:
WTF::Vector

Reply:
Does this function remove the constraints from the passed-in view? If not, I would name it "copy" rather than "transfer".

Reply:
Hi @geoffreygaren - thanks for the extensive review and continuing education about `RetainPtr`... I'll take another pass to fix those.

To answer the question here about whether the constraints are removed or copied, they are removed. The line "oldConstraint.active = NO;" removes the constraint from the view it was previously attached to.